### PR TITLE
Fix part of #138: Topic train fragment Low-fi UI (Part 2)

### PR DESCRIPTION
**Pull request by rt4914**
## Explanation

Replicated from #197 

Part 1: #200 
Part 2: This PR contains the skill item implementation which will be used inside recyclerview. 
Part 3: #203
Part 4: #204

Mock: https://xd.adobe.com/spec/e2239cf4-9cde-4c08-5296-25316c1f0a14-9412/screen/f403add1-079a-4202-9975-efe04c076290/Home-Page-1-Start-Learning-Pre-Scroll-6

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is **assigned** to an appropriate reviewer.

### CODE REVIEW

**Comment by rt4914** [comment]
@veena14cs @nikitamarysolomanpvt PTAL

File: app/src/main/java/org/oppia/app/topic/train/SkillSelectionAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import androidx.databinding.DataBindingUtil
+import androidx.databinding.ViewDataBinding
+import androidx.databinding.library.baseAdapters.BR
+import androidx.recyclerview.widget.RecyclerView
+import kotlinx.android.synthetic.main.topic_train_skill_view.view.*
+import org.oppia.app.R
+import org.oppia.app.databinding.TopicTrainSkillViewBinding
+
+// TODO(#172): Make use of generic data-binding-enabled RecyclerView adapter.
+/** Adapter to bind skills to [RecyclerView] inside [TopicTrainFragment]. **/
```
Comment (veena14cs) [inline app/src/main/java/org/oppia/app/topic/train/SkillSelectionAdapter.kt]: Correct this java doc comment. Must ends with */

**Review by veena14cs — APPROVED** [review]
Correct javadoc comment and go ahead.

**Review by nikitamarysolomanpvt — APPROVED** [review]
LGTM

**Review by nikitamarysolomanpvt — APPROVED** [review]
LGTM

File: app/src/main/java/org/oppia/app/topic/train/SkillSelectionAdapter.kt (hunk 1 above)
Comment (rt4914) [reply app/src/main/java/org/oppia/app/topic/train/SkillSelectionAdapter.kt]: Thanks for this nit. Done.

File: app/src/main/java/org/oppia/app/topic/train/SkillInterface.kt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package org.oppia.app.topic.train
+
+/** Interface to update the selectedSkillList in [TopicTrainFragmentPresenter]. */
+interface SkillInterface {
```
Comment (BenHenning) [inline app/src/main/java/org/oppia/app/topic/train/SkillInterface.kt]: Maybe: SkillSelector would be clearer. Using 'interface' in the name feels a bit odd. It's like calling something 'SkillObject'.

File: app/src/main/java/org/oppia/app/topic/train/SkillSelectionAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+
+  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
+    val inflater = LayoutInflater.from(parent.context)
+    val binding =
+      DataBindingUtil.inflate<TopicTrainSkillViewBinding>(
+        inflater,
+        R.layout.topic_train_skill_view,
+        parent,
+        false
```
Comment (BenHenning) [inline app/src/main/java/org/oppia/app/topic/train/SkillSelectionAdapter.kt]: Nit: please add in-line comment with the argument name corresponding to this literal to provide context to readers, e.g.:

/* attachToParent= */ false

**Review by BenHenning — APPROVED** [review]
Thanks @rt4914!

File: app/src/main/java/org/oppia/app/topic/train/SkillInterface.kt (hunk 2 above)
Comment (rt4914) [reply app/src/main/java/org/oppia/app/topic/train/SkillInterface.kt]: Actually your comment makes sense. Updated `SkillInterface` to `SkillSelector`.

File: app/src/main/java/org/oppia/app/topic/train/SkillSelectionAdapter.kt (hunk 3 above)
Comment (rt4914) [reply app/src/main/java/org/oppia/app/topic/train/SkillSelectionAdapter.kt]: Done.

**Comment by rt4914** [comment]
Merging after resolving all the conversations.

**Pull request merged**